api/src/lib.rs

```diff
@@ -871,16 +944,31 @@ where
     }
 
     {
+        // Magic shares
         let mut shares =
             tide::with_state(State::new(share_db.clone(), token_db.clone()));
         shares
             .at(":id")
-            .with(cors)
             .with(ensure_share_id)
             .get(fetch_share)
             .post(store_share);
         api.at("shares").nest(shares);
     }
 
+    {
+        // Verification tokens
+        let mut verify_tokens =
+            tide::with_state(State::new(verify_db.clone(), token_db.clone()));
+        verify_tokens
+            .at(":id")
+            .with(add_auth_info)
+            .with(signed_pow_auth)
```

Comment:
we're going to have to figure out how to avoid the costly pow for token creation here. we want the request signed, agree. but we need it to not be too expensive to verify your email

Reply:
It would be nice to be able to sign requests separately from signing proof of work.

Reply:
yeah the move I've been thinking about it the more I think we need to pull the two things out